### ISSUE TITLE
Compile for Varnish 4.0

### ISSUE THREAD

**Issue by razvanphp** [varnish/libvmod-shield#2]
I tried to port this vmod, looking at examples from other vmods, `varnish.m4` file is found, but I get a different error:

````
make  all-recursive
make[1]: Entering directory `/home/r/libvmod-shield'
Making all in src
make[2]: Entering directory `/home/r/libvmod-shield/src'
make[2]: *** No rule to make target `/lib/libvmod_std/vmod.py', needed by `vcc_if.c'.  Stop.
make[2]: Leaving directory `/home/r/libvmod-shield/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/r/libvmod-shield'
make: *** [all] Error 2
````

**Comment by aondio** [comment]
Hi,
this error is related to libvmod-shield/src/Makefile.am, have you ported to 4 this file as well?
it seems like there are some macros not defined properly.

**Comment by razvanphp** [comment]
Yes, thank you, I could move on with that problem but I soon run into others:

````
vmod_shield.c:17:18: error: 'struct sess' has no member named 'step'
vmod_shield.c:17:28: error: 'STP_RECV' undeclared (first use in this function)
vmod_shield.c:17:28: note: each undeclared identifier is reported only once for each function it appears in
vmod_shield.c:23:23: error: 'params' undeclared (first use in this function)
vmod_shield.c:26:2: error: implicit declaration of function 'VTCP_linger' [-Werror=implicit-function-declaration]
vmod_shield.c:28:2: error: implicit declaration of function 'vca_close_session' [-Werror=implicit-function-declaration]
````

I tried to replace the `sp` variable but I can't seem to make it work...

**Comment by aondio** [comment]
Hi,
Header files: remove bin/varnishd/cache.h and add cache/cache.h.
'sp' is now 'vrt_ctx->req'.

There is this nice blog describing how to port a vmod from 3 to 4: https://lassekarstensen.wordpress.com/2013/12/19/converting-a-varnish-3-0-vmod-to-4-0/

**Comment by razvanphp** [comment]
@aondio Sorry, I tried, but I'm not a C developer so I can't make it work, I don't have enough knowledge to make it even compile, not to say work.

Can you please take the time to port this for varnish 4? It's a rather simple and useful vmod for DDoS protection.

Thank you!

**Comment by blindpet** [comment]
I am also interested in using this with Varnish 4. I'm no developer but I can follow tutorials. Can you elaborate on the blog entry explaining how to do it, for example where are these header files?

Is there somebody I can donate to to make this happen?

**Comment by aondio** [comment]
Hi, I've just ported the VMOD to V4.
Use branch 4.0 and build it against your Varnish4.
Feel free to raise a new issue if errors should occur.
